### PR TITLE
Clearing app_offline.htm that may be leftover

### PR DESCRIPTION
**Pull request by ankitkumarr**
I need to test this in a private site.
Also need to fix tests.

### CODE REVIEW

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 1]
```diff
@@ -740,5 +741,33 @@ public static async Task UpdateSiteVersion(ZipDeploymentInfo deploymentInfo, IEn
...
+            var appOfflineFile = Path.Combine(environment.WebRootPath, Constants.AppOfflineFileName);
+            if (FileSystemHelpers.FileExists(appOfflineFile))
+            {
+                if (deploymentLock.IsHeld)
+                {
+                    tracer.Trace($"Deployment lock is held, will not remove {Constants.AppOfflineFileName}");
+                    return;
+                }
+
+                try
```
Comment (suwatch) [inline Kudu.Core/Helpers/PostDeploymentHelper.cs]: I would try/catch the whole thing to be saved.

Comment (ankitkumarr) [reply Kudu.Core/Helpers/PostDeploymentHelper.cs]: Good point! I meant that have the callers use "SafeOperation". But catching the whole thing and logging is probably a better option.

File: Kudu.Core/Helpers/PostDeploymentHelper.cs [hunk 2]
```diff
@@ -740,5 +741,33 @@ public static async Task UpdateSiteVersion(ZipDeploymentInfo deploymentInfo, IEn
...
+                if (FileSystemHelpers.FileExists(appOfflineFile))
+                {
+                    if (deploymentLock.IsHeld)
+                    {
+                        tracer.Trace($"Deployment lock is held, will not remove {Constants.AppOfflineFileName}");
+                        return;
+                    }
+
+                    var appOfflineContent = OperationManager.Attempt(() => FileSystemHelpers.ReadAllText(appOfflineFile));
+                    if (appOfflineContent.Contains(Constants.AppOfflineKuduContent))
```
Comment (ankitkumarr) [inline Kudu.Core/Helpers/PostDeploymentHelper.cs]: can make this ignore case. Not a big deal.

**Comment by suwatch** [comment]
merged commit 91aa9708

**Pull request closed without merging**